{"hierarchy":{"paths":[["doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit"]]},"primaryContentSections":[{"declarations":[{"tokens":[{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"kind":"identifier","text":"apptentiveAssertionHandler"},{"kind":"text","text":": ("},{"kind":"attribute","text":"@autoclosure "},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:s12StaticStringV","text":"StaticString"},{"kind":"text","text":", "},{"preciseIdentifier":"s:Su","kind":"typeIdentifier","text":"UInt"},{"text":") -> ()","kind":"text"}],"languages":["swift"],"platforms":["iOS"]}],"kind":"declarations"},{"kind":"content","content":[{"text":"Discussion","level":2,"type":"heading","anchor":"discussion"},{"inlineContent":[{"text":"This can be overriden, for example:","type":"text"}],"type":"paragraph"},{"code":["apptentiveAssertionHandler = { message, file, line in","    print(\"\\(file):\\(line): Apptentive critical error: \\(message())\")","}"],"type":"codeListing","syntax":null}]}],"schemaVersion":{"minor":3,"major":0,"patch":0},"metadata":{"fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"apptentiveAssertionHandler","kind":"identifier"},{"kind":"text","text":": (() -> "},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"kind":"text","text":", "},{"text":"StaticString","kind":"typeIdentifier","preciseIdentifier":"s:s12StaticStringV"},{"kind":"text","text":", "},{"text":"UInt","kind":"typeIdentifier","preciseIdentifier":"s:Su"},{"text":") -> ()","kind":"text"}],"modules":[{"name":"ApptentiveKit"}],"role":"symbol","roleHeading":"Global Variable","title":"apptentiveAssertionHandler","externalID":"s:13ApptentiveKit26apptentiveAssertionHandleryySSyXK_s12StaticStringVSutcvp","symbolKind":"var"},"abstract":[{"type":"text","text":"The method to call when a critical error occurs."}],"sections":[],"kind":"symbol","variants":[{"paths":["\/documentation\/apptentivekit\/apptentiveassertionhandler"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit\/apptentiveAssertionHandler"},"references":{"doc://com.apptentive.Apptentive/documentation/ApptentiveKit/apptentiveAssertionHandler":{"identifier":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit\/apptentiveAssertionHandler","type":"topic","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"apptentiveAssertionHandler"},{"text":": (() -> ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"StaticString","kind":"typeIdentifier","preciseIdentifier":"s:s12StaticStringV"},{"kind":"text","text":", "},{"kind":"typeIdentifier","preciseIdentifier":"s:Su","text":"UInt"},{"text":") -> ()","kind":"text"}],"kind":"symbol","abstract":[{"text":"The method to call when a critical error occurs.","type":"text"}],"role":"symbol","url":"\/documentation\/apptentivekit\/apptentiveassertionhandler","title":"apptentiveAssertionHandler"},"doc://com.apptentive.Apptentive/documentation/ApptentiveKit":{"kind":"symbol","identifier":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit","type":"topic","title":"ApptentiveKit","url":"\/documentation\/apptentivekit","abstract":[],"role":"collection"}}}